src/App.tsx

```diff
@@ -15,48 +17,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  None = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  reverseField = false,
+) {
+  const visibleGoods = [...goods].map((good, index) => (
+    {
+      name: good,
+      id: (index + 1),
+      long: good.length,
+    }
+  ));
```

Comment:
Why complicate the code so much? Here it is enough to make an ordinary copy
```suggestion
  const prepearedGoods = [...goods];
```